test/yorkie_test.ts

```diff
@@ -430,4 +431,124 @@ describe('Yorkie', function () {
+      d2.update((root) => {
+        root['k1'] = 'undefined';
+      });
+
+      await c2.sync();
+      await c1.sync();
+      assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
+
+      // Simulate network error
+      const xhr = sinon.useFakeXMLHttpRequest();
+      xhr.onCreate = (req) => {
+        req.respond(400, {
+          'Content-Type': 'application/grpc-web-text+proto',
+        }, null);
+      };
+
+      d2.update((root) => {
+        root['k1'] = 'v1';
+      });
+
+      await c2.sync().catch(err => {
+        assert.equal(err.message, 'INVALID_STATE_ERR - 0')
+      });
+      await c1.sync().catch(err => {
+        assert.equal(err.message, 'INVALID_STATE_ERR - 0')
+      });
+      assert.equal(d1.toSortedJSON(), '{"k1":"undefined"}');
+      assert.equal(d2.toSortedJSON(), '{"k1":"v1"}');
+
+      // Back to normal condition
+      xhr.restore();
+
+      await c2.sync();
+      await c1.sync();
+      assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
+    }, this.test.title);
+  });
+
+  it('Can recover from temporary disconnect (realtime sync)', async function() {
+    const c1 = yorkie.createClient(testRPCAddr);
+    const c2 = yorkie.createClient(testRPCAddr);
+    await c1.activate();
+    await c2.activate();
+
+    const docKey = `${this.test.title}-${new Date().getTime()}`;
+    const d1 = yorkie.createDocument(testCollection, docKey);
+    const d2 = yorkie.createDocument(testCollection, docKey);
+
+    await c1.attach(d1);
+    await c2.attach(d2);
+
+    const listener1 = new EventEmitter();
+    const listener2 = new EventEmitter();
+    const createSpy = (emitter: EventEmitter) => {
+      return (event: ClientEvent | DocEvent) => emitter.emit(event.name);
+    };
+    const spy1 = createSpy(listener1);
+    const spy2 = createSpy(listener2);
+    const waitFor = (eventName: string, listener: EventEmitter) => {
+      return new Promise(resolve => listener.on(eventName, resolve));
+    };
+
+    const unsub1 = {
+      client: c1.subscribe(spy1),
+      doc: d1.subscribe(spy1),
+    };
+    const unsub2 = {
+      client: c2.subscribe(spy2),
+      doc: d2.subscribe(spy2),
+    };
+
+    // Normal Condition
+    d2.update((root) => {
+      root['k1'] = 'undefined';
+    });
+
+    await waitFor(DocEventType.LocalChange, listener2);  // d2 should be able to update
+    await waitFor(DocEventType.RemoteChange, listener1); // d1 should be able to receive d2's update
+    assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
+
+    // Simulate network error
+    const xhr = sinon.useFakeXMLHttpRequest();
+    xhr.onCreate = (req) => {
+      req.respond(400, {
+        'Content-Type': 'application/grpc-web-text+proto',
+      }, null);
+    };
+
+    d2.update((root) => {
+      root['k1'] = 'v1';
+    });
+
+    await waitFor(DocEventType.LocalChange, listener2);  // d2 should be able to update
+    await waitFor(ClientEventType.SyncFailed, listener2); // c2 should fail to sync
+    c1.sync();
```

Comment:
Because we've simulated network error at this point, `c1` does not notice any remote changes. Therefore `c1` would not try to sync in the `runSyncLoop()`, which is why we need to manually attempt to sync `c1` at this point.